src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/HandleNotRevertedSubFlowAction.java

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.fsm.yflow.update.action;
+
+import static java.lang.String.format;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.wfm.topology.flowhs.fsm.common.actions.NbTrackableAction;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateContext;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateFsm.Event;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateFsm.State;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+public class HandleNotRevertedSubFlowAction
+        extends NbTrackableAction<YFlowUpdateFsm, State, Event, YFlowUpdateContext> {
+    public HandleNotRevertedSubFlowAction(PersistenceManager persistenceManager) {
+        super(persistenceManager);
+    }
+
+    @Override
+    protected Optional<Message> performWithResponse(State from, State to, Event event, YFlowUpdateContext context,
+                                                    YFlowUpdateFsm stateMachine) {
+        String flowId = context.getSubFlowId();
+
+        if (stateMachine.isFailedSubFlow(flowId)) {
+            log.debug("Ignore an error event for already failed sub-flow " + flowId);
+
+            return Optional.empty();
+        }
+
+        if (!stateMachine.isPendingSubFlow(flowId)) {
+            throw new IllegalStateException("Received an event for non-pending sub-flow " + flowId);
+        }
+
+        stateMachine.saveErrorToHistory("Failed to revert sub-flow",
+                format("Failed to revert sub-flow %s of y-flow %s: %s", flowId, stateMachine.getYFlowId(),
+                        context.getError()));
+
+        stateMachine.addFailedSubFlow(flowId);
```

Comment:
I've changed the sequence of operations in the similar method of YFlowCreateFsm due to a code review comment. Please, check and align.

Reply:
Fixed.